src/api/Flooq.IntegrationTest/DataFlowTest.cs

```diff
@@ -0,0 +1,151 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Flooq.Api.Models;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Newtonsoft.Json;
+using NuGet.Protocol;
+using Factory = Flooq.IntegrationTest.FlooqWebApplicationFactory<Program>;
+
+namespace Flooq.IntegrationTest;
+
+[TestClass]
+public class DataFlowTest
+{
+  private HttpClient _client;
+  
+  [TestInitialize]
+  public void Setup()
+  {
+    _client = Factory.Factory.CreateClient();
+  }
+  
+  [TestMethod]
+  public async Task CanGetDataFlows()
+  {
+    var response = await _client.GetAsync("api/DataFlow");
+    response.EnsureSuccessStatusCode();
+    
+    var content = response.Content.ReadAsStringAsync().Result;
+    var dataFlows = JsonConvert.DeserializeObject<IEnumerable<DataFlow>>(content)!;
+    
+    Assert.IsFalse(dataFlows.ToImmutableList().IsEmpty);
+  }
+
+  [TestMethod]
+  public async Task CanGetDataFlow()
+  {
+    var response = await _client.GetAsync($"api/DataFlow/{Factory.TEST_GUID}");
+    response.EnsureSuccessStatusCode();
+    
+    var content = response.Content.ReadAsStringAsync().Result;
+    var dataFlow = JsonConvert.DeserializeObject<DataFlow>(content)!;
+    
+    Assert.AreEqual(Factory.TEST_GUID, dataFlow.Id);
+  }
+
+  [TestMethod]
+  public async Task CannotGetNonExistingDataFlow()
+  {
+    var response = await _client.GetAsync($"api/DataFlow/{Guid.NewGuid()}");
+
+    Assert.AreEqual(HttpStatusCode.NotFound, response.StatusCode);
+  }
+
+  [TestMethod]
+  public async Task CanPostDataFlow()
+  {
+    var dataFlow = new DataFlow()
+    {
+      Id = Guid.NewGuid(),
```

Comment:
`Id` and `LastEdited` should be set by the API after the refactoring from @silvan-flum. So you don't need to set these fields but you should test if they are set correctly by the api.

Reply:
`Id` works fine, but `LastEdited` is null. With adding ``dataFlow.LastEdited = DateTime.UtcNow;`` in DataFlowController.PostDataFlow it works. @silvan-flum What about that?